keps/sig-scheduling/5471-enable-sla-based-scheduling/README.md

```diff
@@ -0,0 +1,914 @@
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+Potentially yes, but the impact should be **minimal**. The numeric toleration operators feature could slightly increase time for operations covered by existing SLIs/SLOs due to integer parsing overhead and validation overhead.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+Same as existing taint/toleration system which is graceful degradation.
+
+###### What are other known failure modes?
+
+A failure mode due to numeric toleration operators have integer parsing errors from malformed taint/toleration values causing pods to be rejected with clear error messages.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+N/A
+
+## Implementation History
+
+- 2025-08-11: Initial KEP
+
+## Drawbacks
+
+## Alternatives
+
+There are many different alternatives were considered:
+
+1. **New Dedicated SLA API Resource:**  Create `SLAPolicy` CRD
+   - **Pros:** Clean separation, rich policy definitions.
+   - **Cons:** New API surface, additional complexity, breaks unified taint/toleration model.
+2. **Custom Scheduler Plugin:** Use scheduling plugin with SLA-aware logic, [placement-policy-scheduler-plugins](https://github.com/Azure/placement-policy-scheduler-plugins)
+   - **Pros:** Full scheduling control, rich logic possible
+   - **Cons:**
+     - Out-of-tree scheduler plugin to maintain and manage
+     - Doesn't leverage existing taint/toleration infrastructure.
+3. **Node Labels + Enhanced NodeAffinity:** Use labels instead of taints, extend NodeAffinity matching.
+   - **Pros:** Leverages existing label system.
+   - **Cons:**
+     - No default push-back behavior
+     - No eviction semantics
+     - Labels aren't meant for operational constraints.
+
+4. **Add Separate `NumValue int64` Field:** Add a dedicated numeric field alongside the existing `Value string` field in Taint/Toleration structs.
+   - **Pros:**
+     - Eliminates parsing overhead and errors
+     - Type-safe integer handling
+     - No concerns about leading zeros or malformed values
+     - Better performance for numeric comparisons
+   - **Cons:**
+     - Not aesthetically pleasing API design with dual fields
+     - Users might set wrong field or both fields accidentally
+     - Complex validation logic for field combinations
+     - Memory/storage overhead for additional field
+     - API complexity and documentation burden
```

Comment:
I'll play devil's advocate, have you considered using the current mechanism such that it only works based on existing operators? Iow. Node can publish `node.kubernetes.io/sla=950`, and pods will just use `sla equal 950`. What are the pros and cons of such approach?